examples/gno.land/r/stefann/config/config.gno

```diff
@@ -0,0 +1,53 @@
+package config
+
+import (
+	"errors"
+	"std"
+)
+
+var (
+	main   std.Address
+	backup std.Address
+)
+
+func init() {
+	main = "g1sd5ezmxt4rwpy52u6wl3l3y085n8x0p6nllxm8"
+	backup = "g13awn2575t8s2vf3svlprc4dg0e9z5wchejdxk8"
+}
+
+func Address() std.Address {
+	return main
+}
+
+func Backup() std.Address {
+	return backup
+}
+
+func SetAddress(a std.Address) error {
+	if !a.IsValid() {
+		return errors.New("config: invalid address")
+	}
+
+	if err := checkAuthorized(); err != nil {
+		return err
+	}
+
+	main = a
+	return nil
+}
+
+func checkAuthorized() error {
+	caller := std.GetOrigCaller()
```

Comment:
You should switch to `std.PrevRealm.Addr()` to prevent someone from creating an intermediary contract that imports yours and calls `SetAddress()`, which could result in the theft of your ownership.

Reply:
I explained the solution chose and the thought process behind it in the comment below. https://github.com/gnolang/gno/pull/2918#discussion_r1796887236